tests/DwitTech.AccountService.WebApi.Tests/Controllers/UserControllerTest.cs

```diff
@@ -353,5 +353,81 @@ public async Task Logout_Should_Return_False_If_AuthorizationHeader_Is_Not_Given
             Assert.IsType<BadRequestObjectResult>(result);
 
         }
+
+        [Fact]
+        public async Task DeleteUser_Returns_NoContentResult()
+        {
+            // Arrange
+            var userId = 1;
+            var _activationService = new Mock<IActivationService>();
+            var iloggerMock = new Mock<ILogger<UserController>>();
+            var _mockAuthService = new Mock<IAuthenticationService>();
+            var _mockUserService = new Mock<IUserService>();
+
+            var userController = new UserController(_activationService.Object, _mockAuthService.Object, _mockUserService.Object, iloggerMock.Object);
+
+            // Act
+            var result = await userController.DeleteUser(userId);
+
+            // Assert
+            Assert.IsType<NoContentResult>(result);
+            _mockUserService.Verify(service => service.DeleteUserAsync(userId), Times.Once);
+            iloggerMock.Verify(
+                logger => logger.Log(
+                    LogLevel.Information,
+                    It.IsAny<EventId>(),
+                    It.Is<It.IsAnyType>((o, t) => o.ToString().Contains($"User with ID {userId} deleted")),
+                    It.IsAny<Exception>(),
+                    (Func<It.IsAnyType, Exception, string>)It.IsAny<object>()),
+                Times.Once);
+        }
+
+        [Fact]
+        public async Task DeleteUser_WithInvalidId_ReturnsNotFound()
+        {
+            // Arrange
+            var id = 1;
+            var exceptionMessage = $"User with ID {id} not found";
+            var _activationService = new Mock<IActivationService>();
+            var iloggerMock = new Mock<ILogger<UserController>>();
+            var _mockAuthService = new Mock<IAuthenticationService>();
+            var _mockUserService = new Mock<IUserService>();
+
+            var userController = new UserController(_activationService.Object, _mockAuthService.Object, _mockUserService.Object, iloggerMock.Object);
+
+            _mockUserService
+                    .Setup(service => service.DeleteUserAsync(id))
+                    .Throws(new ArgumentException($"User with ID {id} not found"));
+            // Act
+            var result = await userController.DeleteUser(id);
+
+            // Assert
+            var notFoundResult = Assert.IsType<NotFoundObjectResult>(result);
+            Assert.Equal("User with ID 1 not found", notFoundResult.Value);
+        }
+
+        [Fact]
+        public async Task DeleteUser_WithException_ReturnsStatusCode500()
+        {
+            // Arrange
+            var id = 1;
+            var errorMessage = "Something went wrong";
```

Comment:
This variable was not used in the test. Perhaps there should be an assert statement to verify that the message in the variable matches the message generated in the act.

Reply:
> This variable was not used in the test. Perhaps there should be an assert statement to verify that the message in the variable matches the message generated in the act.

Resolved